### ISSUE TITLE
MacVim bookmark error 

### ISSUE THREAD

**Issue by deltaskelta** [scrooloose/nerdtree#677]
MacVim: Version 8.0.502 (129)
NERDTree: 5.0.0

After making a bookmark of a node I get this error when I am trying to open the bookmark...

```
Error detected while processing function nerdtree#ui_glue#openBookmark[2]..87:
line    2:
E121: Undefined variable: nerdtree
E116: Invalid arguments for function 86
E15: Invalid expression: bookmark.getNode(nerdtree, a:searchFromAbsoluteRoot)

```

**Comment by lifecrisis** [comment]
Hi, @deltaskelta. Thanks for bringing this to our attention.

This looks like a bug that should have surfaced quite some time ago. I have a fix for you to try...

```patch
From 953fa16ffe14cc38c9f558043ca4b238a71d1fda Mon Sep 17 00:00:00 2001
From: Jason Franklin <j_fra@fastmail.us>
Date: Fri, 2 Jun 2017 19:49:51 -0400
Subject: [PATCH] Add a missing function argument sigil

An error message from issue #677 hinted that an argument sigil was
missing in a function in the "bookmark.vim" script. It is unknown why
this bug did not surface until now, but this commit puts the simple fix
in place.

Fixes #677.
---
 lib/nerdtree/bookmark.vim | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/lib/nerdtree/bookmark.vim b/lib/nerdtree/bookmark.vim
index ad8c3d1..e9479ac 100644
--- a/lib/nerdtree/bookmark.vim
+++ b/lib/nerdtree/bookmark.vim
@@ -163,7 +163,7 @@ endfunction
 " treenode for it.
 function! s:Bookmark.GetNodeForName(name, searchFromAbsoluteRoot, nerdtree)
     let bookmark = s:Bookmark.BookmarkFor(a:name)
-    return bookmark.getNode(nerdtree, a:searchFromAbsoluteRoot)
+    return bookmark.getNode(a:nerdtree, a:searchFromAbsoluteRoot)
 endfunction
 
 " FUNCTION: Bookmark.GetSelected() {{{1
-- 
2.7.4

```

Apply the above patch to your local NERDTree repository (since the change is small, you could just do it manually).  If it works for you, I'll submit a pull request to @PhilRunninger to make the change public.

**Comment by deltaskelta** [comment]
you know, I tried to go through the bookmark process again and it works without failing. I am not sure why this happened before, maybe it was the older version of MacVim that did this. I am now on `8.0.596` and it works without the patch. Thanks for the help!

...Closing

**Comment by lifecrisis** [comment]
Hey, @deltaskelta. This issue should actually **not** be closed.

I was able to identify and isolate the issue you were having. I can easily replicate the behaviour now. I don't have time to do everything I want right away, and, since this bug isn't critical, I'll likely wait until this weekend to submit a pull request.

Could you reopen?

@PhilRunninger, could you reopen this if he doesn't see it before you? It's an easy fix... and the patch above would do the magic.  However, I'd prefer to wait a bit and include a few more commits to clean up some related code that has grown a tad stale. I'll have more to say about the exact nature of the bug in my PR. Does that sound okay?

**Comment by PhilRunninger** [comment]
Sounds good. I'm curious to learn how to replicate the error.

**Comment by lifecrisis** [comment]
Hey guys, details for the fix can be found in pull request #703. Thanks for bringing this up, @deltaskelta.
